src/UserService.Business/Commands/User/EditUserCommand.cs

```diff
@@ -263,41 +320,46 @@ public OperationResultResponse<bool> Execute(Guid userId, JsonPatchDocument<Edit
             if (imageOperation != null)
             {
                 imageId = GetAvatarImageId(JsonConvert.DeserializeObject<AddImageRequest>(imageOperation.value?.ToString()), errors);
-                if (imageId == null)
-                {
-                    status = OperationResultStatusType.PartialSuccess;
-                }
             }
 
             if (positionOperation != null)
             {
-                if (!ChangeUserPosition(Guid.Parse(positionOperation.value.ToString() ?? string.Empty), userId, errors))
-                {
-                    status = OperationResultStatusType.PartialSuccess;
-                }
+                ChangeUserPosition(Guid.Parse(positionOperation.value.ToString() ?? string.Empty), userId, errors);
             }
 
             if (departmentOperation != null)
             {
-                if (!ChangeUserDepartment(Guid.Parse(departmentOperation.value.ToString() ?? string.Empty), userId, errors))
-                {
-                    status = OperationResultStatusType.PartialSuccess;
-                }
+                ChangeUserDepartment(Guid.Parse(departmentOperation.value.ToString() ?? string.Empty), userId, errors);
             }
 
             if (roleOperation != null)
             {
-                if (!ChangeUserRole(Guid.Parse(roleOperation.value.ToString() ?? string.Empty), userId, errors))
-                {
-                    status = OperationResultStatusType.PartialSuccess;
-                }
+                ChangeUserRole(Guid.Parse(roleOperation.value.ToString() ?? string.Empty), userId, errors);
             }
 
             if (officeOperation != null)
             {
-                if (!ChangeUserOffice(Guid.Parse(officeOperation.value.ToString() ?? string.Empty), userId, errors))
+                ChangeUserOffice(Guid.Parse(officeOperation.value.ToString() ?? string.Empty), userId, errors);
+            }
+
+            if (isActiveOperation != null)
+            {
+                bool dbUserIsActive = _userRepository.Get(userId).IsActive;
+
+                if (bool.Parse(isActiveOperation.value.ToString()) && !dbUserIsActive)
+                {
+                    string password = _generatePassword.Execute();
+
+                    _userRepository.CreatePending(new() { UserId = userId, Password = password });
```

Comment:
why we create new password if user has old password

Reply:
it is not safe to use old password